flaml/online_automl/auto_vw.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+from typing import Optional
+import logging
+from ..tune.trial import Trial
+from ..tune.online_trial_runner import OnlineTrialRunner
+from ..scheduler.online_scheduler import ChaChaScheduler
+from ..searcher.online_searcher import ChampionFrontierSearcher
+logger = logging.getLogger(__name__)
+
+
+class AutoVW:
+    """Implemention of AutoVW
+
+    Args:
+        ************Args about the task***************
+        problem: problem, which is instance of VWNameSpaceInteraction. It provides task related info: 
+                 init_config, search_space, and trainable_func, for example an VW instance.
+                 (Side note about the problem: the dataset is also part of problem, so the problem
+                 also knows dataset related info, e.g. dim of feature)
+        concurrent_running_budget (int): the max number of live models allowed. Default = 5
+        ************Args about ChaCha***************
+        max_live_model_num (int): the maximum number of models allowed
+        min_resource_lease (float or str): if set as 'auto', it will be calculated automatically
+        automl_runner_args (dict): configuration for the OnlineTrialRunner
+        scheduler_args (dict): configuration for the scheduler
+        model_select_policy (str): how to select one model to do prediction from the live model pool
+        metric (str): loss function used for calculating the progressive validation loss
+        config_oracle_random_seed (int): random seed for the ConfigOracle
+        cb_coef (float): an optional coefficient in the sample complexity bound
+    """
+    MIN_RES_CONST = 5
+    WARMSTART_NUM = 100
+
+    def __init__(self,
+                 problem,
+                 max_live_model_num: int,
+                 min_resource_lease='auto',
+                 automl_runner_args: dict = {},
+                 scheduler_args: dict = {},
+                 model_select_policy: str = 'threshold_loss_ucb',
+                 metric='mae_clipped',
+                 config_oracle_random_seed: Optional[int] = None,
+                 model_selection_mode='min',
+                 cb_coef: Optional[float] = None,
+                 ):
+        self._max_live_model_num = max_live_model_num
```

Comment:
docstr should be placed here. We shouldn't require dataset as part of the input to the constructor. Think about AutoVW as a VW learner.